tests/utils.py

```diff
@@ -45,3 +46,46 @@ def assert_files_equal(file: Path, reference_file: Path, skip_lines: list = []):
         actual = actual_lines[i]
         reference = reference_lines[i]
         assert actual == reference, f"<{actual}> not equal to <{reference}>"
+
+
+def assert_file_is_same_binary(
+    input_directory: Path,
+    input_filepath: Optional[Path],
+    reference_directory: Path,
+    reference_filepath: Path = ...,
+) -> None:
+    """Assert that input_directory / input_filepath is equal to
+    reference_directory / reference_filepath
+
+    If input_filepath is None, nothing will be compared.
+
+    If no reference_filepath is specified it will be set to input_filepath.
+
+    If the reference_directory / reference_filepath does not exist, we assert
+    that input_directory / input_filepath does not exist.
+
+    Args:
+        input_directory (Path):
+            The directory containing the input file.
+        input_filepath (Optional[Path]):
+            The input filename relative to input_directory
+        reference_directory (Path):
+            The directory containing the reference file.
+        reference_filepath (Path, optional):
+            The reference filename relative to reference_directory
+            Defaults to ... which is then set to input_filepath.
+    """
+    if input_filepath is None:
+        return
+    if reference_filepath is ...:
```

Comment:
Would this not match *anything*? (whereas you mean None/empty string?)

Reply:
All clear, I overlooked the default argument. Thanks for explaining, review ok now.